specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2023-02-01-preview/containerInstance.json

```diff
@@ -0,0 +1,2463 @@
+        },
+        "periodSeconds": {
+          "description": "The period seconds.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "failureThreshold": {
+          "description": "The failure threshold.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "successThreshold": {
+          "description": "The success threshold.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "timeoutSeconds": {
+          "description": "The timeout seconds.",
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "ContainerGroup": {
+      "description": "A container group.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        },
+        {
+          "$ref": "#/definitions/ContainerGroupProperties"
+        }
+      ]
+    },
+    "ContainerGroupProperties": {
+      "description": "The container group properties",
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/ContainerGroupIdentity",
+          "description": "The identity of the container group, if configured."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "description": "The container group properties",
+          "properties": {
+            "provisioningState": {
+              "readOnly": true,
+              "type": "string",
+              "description": "The provisioning state of the container group. This only appears in the response."
+            },
+            "containers": {
+              "type": "array",
+              "description": "The containers within the container group.",
+              "items": {
+                "$ref": "#/definitions/Container"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "imageRegistryCredentials": {
+              "type": "array",
+              "description": "The image registry credentials by which the container group is created from.",
+              "items": {
+                "$ref": "#/definitions/ImageRegistryCredential"
+              },
+              "x-ms-identifiers": [
+                "username"
+              ]
+            },
+            "provisioningTimeoutInSeconds": {
+              "type": "integer",
+              "description": "Time in seconds in which a container group deployment would timeout and fail. The allowed maximum value is 1800 seconds. If value is not provided, property is given maximum value by default.",
+              "format": "int32"
+            },
+            "isCustomProvisioningTimeout": {
+              "type": "boolean",
+              "readOnly": true,
+              "description": "Flag indicating whether a custom value was provided for the provisioningTimeoutInSeconds property"
+            },
```

Comment:
Consider using an enum instead : 

[https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
You should use x-ms-enum with modelAsString=true
https://azure.github.io/autorest/extensions/#x-ms-enum

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/23221/files#r1149391828